lib/r10k/action/runner.rb

```diff
@@ -92,6 +96,36 @@ def settings_from_config(override_path)
 
         results
       end
+
+      def add_credential_overrides(overrides)
+        sshkey_path = @opts[:'private-key']
+        token_path = @opts[:'oauth-token']
+
+        if sshkey_path && token_path
+          raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
+        end
+
+        if sshkey_path
+          overrides[:git] ||= {}
+          overrides[:git][:private_key] = sshkey_path
+          if repo_settings = overrides[:git][:repositories]
+            repo_settings.each do |repo|
+              repo[:private_key] = sshkey_path
+            end
+          end
+        elsif token_path
+          overrides[:git] ||= {}
+          overrides[:git][:private_key] = sshkey_path
```

Comment:
why are we setting this to nil here?

Reply:
I feel like this might be a typo. It will ignore whichever one isn't needed for the kind of URL associated with each remote, so it's not an issue to have both set. Thanks for catching this.